tests/test_emulator.cpp

```diff
@@ -629,6 +629,24 @@ TEST_F(EmulatorTest, insn_bt) {
         { 0x80000000'00000000ULL, 0xFF, 0,
           0x80000000'00000000ULL, RFLAGS_CF },
         });
+
+    // Variant `bt mN,rN`: Modifies the EA based on the bit offset.
+    test_cpu_t vcpu_original;
+    test_cpu_t vcpu_expected;
+    vcpu_original = {};
+    vcpu_original.gpr[REG_RCX] = 0ULL;
+    (uint64_t&)vcpu_original.mem[0x00] = 0x00020000'00000000;
+    (uint64_t&)vcpu_original.mem[0x08] = 0x00000000'00000000;
+    (uint64_t&)vcpu_original.mem[0x10] = 0xFFFFFFFF'FFFFFFFE;
+    
+    vcpu_original.gpr[REG_RAX] = -15LL;
+    vcpu_expected = vcpu_original;
+    vcpu_expected.flags |= RFLAGS_CF;
+    run("bt [ecx + 0x08], eax", vcpu_original, vcpu_expected);
```

Comment:
I was surprised that this test case passes, due to the flaw I've pointed out above. I inspected `em_ctxt.dst.mem.ea` after `em_emulate_insn()`, and saw the wrong value I had expected (`0x20000006`, instead of `0x07`). Note that this address is well outside the bounds of the `test_cpu_t::mem` array. Then I realized that `test_read_memory()` does `ea &= 0xFF` before validating `ea` and `size`... That has to be fixed.